lib/react_client/react_interop.dart

```diff
@@ -57,39 +57,43 @@ abstract class React {
 ///     }
 ///
 /// Learn more: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
-Ref<CurrentType> createRef<CurrentType>() {
-  return new Ref<CurrentType>();
+Ref<T> createRef<T>() {
```

Comment:
#supernit

The example in the doc comment for `createRef` shows left-side typing like this:

```dart
final Ref<BarComponent> barRef = createRef();
```

When I believe best practice would be something like:

```dart
final barRef = createRef<BarComponent>();
```

Reply:
https://dart.dev/guides/language/effective-dart/design#prefer-type-annotating-public-fields-and-top-level-variables-if-the-type-isnt-obvious

> When in doubt, add a type annotation. Even when a type is obvious, you may still wish to explicitly annotate. If the inferred type relies on values or declarations from other libraries, you may want to type annotate your declaration so that a change to that other library doesn’t silently change the type of your own API without you realizing.

Fields are kind of a grey area, and we tend toward annotating them in our codebases, so that's why I went with this typing.